tools/Azure.Mcp.Tools.EventGrid/src/Commands/Events/EventsPublishCommand.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Net;
+using Azure.Mcp.Core.Extensions;
+using Azure.Mcp.Core.Models.Option;
+using Azure.Mcp.Tools.EventGrid.Options;
+using Azure.Mcp.Tools.EventGrid.Options.Events;
+using Azure.Mcp.Tools.EventGrid.Services;
+
+namespace Azure.Mcp.Tools.EventGrid.Commands.Events;
+
+public sealed class EventGridPublishCommand(ILogger<EventGridPublishCommand> logger) : BaseEventGridCommand<EventsPublishOptions>
+{
+    private const string CommandTitle = "Publish Events to Event Grid Topic";
+    private readonly ILogger<EventGridPublishCommand> _logger = logger;
+
+    public override string Name => "publish";
+
+    public override string Description =>
+        """
+        Publish custom events to Event Grid topics for event-driven architectures. This tool sends structured event data to 
+        Event Grid topics with schema validation and delivery guarantees for downstream subscribers. Returns publish operation 
+        status. Requires topic, data, and optional schema.
+        """;
+
+    public override string Title => CommandTitle;
+
+    public override ToolMetadata Metadata => new()
+    {
+        Destructive = false,
+        Idempotent = false,
+        OpenWorld = false,
+        ReadOnly = false,
+        LocalRequired = false,
+        Secret = false
+    };
+
+    protected override void RegisterOptions(Command command)
+    {
+        base.RegisterOptions(command);
+        command.Options.Add(OptionDefinitions.Common.ResourceGroup);
+        command.Options.Add(EventGridOptionDefinitions.TopicName);
+        command.Options.Add(EventGridOptionDefinitions.EventData);
+        command.Options.Add(EventGridOptionDefinitions.EventSchema);
+    }
+
+    protected override EventsPublishOptions BindOptions(ParseResult parseResult)
+    {
+        var options = base.BindOptions(parseResult);
+        options.ResourceGroup ??= parseResult.GetValueOrDefault<string>(OptionDefinitions.Common.ResourceGroup.Name);
+        options.TopicName = parseResult.GetValueOrDefault<string>(EventGridOptionDefinitions.TopicName.Name);
+        options.EventData = parseResult.GetValueOrDefault<string>(EventGridOptionDefinitions.EventData.Name);
+        options.EventSchema = parseResult.GetValueOrDefault<string>(EventGridOptionDefinitions.EventSchema.Name);
+        return options;
+    }
+
+    public override async Task<CommandResponse> ExecuteAsync(CommandContext context, ParseResult parseResult)
+    {
+        if (!Validate(parseResult.CommandResult, context.Response).IsValid)
+        {
+            return context.Response;
+        }
+
+        var options = BindOptions(parseResult);
+
+        if (!string.IsNullOrEmpty(options.EventSchema) && !IsValidSchemaType(options.EventSchema))
```

Comment:
I meant to move this into `Validate` function as an override of the base class implementation.

function:
```cs
public override ValidationResult Validate(CommandResult commandResult, CommandResponse? commandResponse = null) {
      var validationResult = base.Validate(commandResult, commandResponse);

      // Already invalid.. don't keep going.
      if (validationResult.IsValid) {
           return validationResult;
      }
      
      if (!string.IsNullOrEmpty(options.EventSchema) && !IsValidSchemaType(options.EventSchema))
      {
            validationResult.IsValue = false;
            // error message etc.
      }

      return validationResult;
}
```

So all the validation logic is contained instead of doing part of it in the base class and part of it in this execute method.